addons/mail/static/src/components/chat_window_manager/chat_window_manager_tests.js

```diff
@@ -1977,6 +1977,46 @@ QUnit.test('focusing a chat window of a chat should make new message separator d
     );
 });
 
+QUnit.test('[technical] unread counter in a chat window is not shown on receiving new transient message [REQUIRE FOCUS]', async function (assert) {
+    // technical as we need to remove focus from text input to avoid `channel_seen` call
```

Comment:
I think it's something @alexkuhn asked for

Reply:
Don't exactly remember what I said, but was wondering why comment below
`// need to remove focus from text area to avoid channel_seen`

And it's to reproduce the crash without fix. Thing is it's very technical to do this to make it crash without fix.
I don't like having tests that contain technical aspects like this.

Yes, in practice it crashes, and if we dig deeper the root cause is user interaction in tests are too simple (user click is not just a `.click()`, there's notably a focus change).

It's out of scope to improve user interaction simulations in tests. So just improving comment below is good enough for me. My mistake to let you have think I wanted to mark this test as "[technical]", which is not entirely true as I said above.

Reply:
> Thing is it's very technical to do this to make it crash without fix.

It's not really technical though, it's our spec for mark as seen that depends on focus that comes from a functional decision.

Reply:
Sorry, meant it looks technical. Rest of comment clearly tells me the root cause is simplistic simulation of click in tests.

Basically what I wanted to say with my comment is that comment is badly worded: only the person that made the test / investigated the bug initially will understand it. So the LOC looks out of place, therefore looks "technical". That’s the main problem I have, and even if that’s not "technical", the comment should be good enough to make this clear for everyone, or it feels as bad as this being technical without justification...

Reply:
So just remove the [technical] or should we also change the comment ? I'm not sure what you're expecting in the comment then alex ?

Reply:
remove the `[technical]` tag in test name, and would change user interaction so that it's a user click, not a keydown "ENTER". As a result, the added `.focus()` would be much easier to get. Could still be worth keeping a comment above explicit `.focus()` change, briefly telling that an actual user click also changes the focus.

Reply:
You cannot use click to send message in chat window ..

Reply:
There is no send button on the chat window, enter must be used. It's just that the user is clicking anywhere else after sending the message. It does not come from any technical reason, it comes from the fact it's the user interaction that we are simulating.

Reply:
So the name of test should reflect that... I mean, it's pretty obvious to me something is weird with test right now, the change of focus looks out of place. Please do the appropriate changes so that it doesn't look off, that the test is clearly intended with that. Could be name of test clearly mentioning message is posted while focus is somewhere else...

Reply:
Yes agree, this is my suggestion since the first message of this chain.

Reply:
I could use some suggestion for test name and test comment I should add (and will make us win some time as I'm pretty sure my suggestions won't be as good as you wish :p )

Reply:
For the title I'd just add what I said originally: "unread counter in a chat window is not shown on receiving new transient message _even when composer is not focused_"

And then you can add a comment why we want to test that specific flux:
"The goal of removing the focus is to ensure the thread is not marked as seen automatically. Indeed that would remove the unread counter no matter what, which is already covered by other tests. The goal of this test is to cover the conditions specific to transient messages, and the conditions from focus would otherwise shadow them."

Reply:
Unresolving because I don't see the change. Did you push it?

Reply:
forgot the --force in my push, sorry ... I'll do the changes here and reopen on 14.0 later